tests/topotests/bgp_vpnv4_per_nexthop_label/test_bgp_vpnv4_per_nexthop_label.py

```diff
@@ -0,0 +1,818 @@
+                    if isinstance(labelList, set):
+                        assert (
+                            storedLabel not in labelList
+                        ), "{0}, {1}, label already detected in a previous entry".format(
+                            router.name, entry
+                        )
+                        labelList.add(storedLabel)
+                else:
+                    assert (
+                        path["remoteLabel"] == storedLabel
+                    ), "{0}, {1}, label value not expected one (expected {2}, observed {3}".format(
+                        router.name, entry, storedLabel, path["remoteLabel"]
+                    )
+                if label_value_expected:
+                    assert (
+                        path["remoteLabel"] == label_value_expected
+                    ), "{0}, {1}, label value not expected (expected {2}, observed {3}".format(
+                        router.name, entry, label_value_expected, path["remoteLabel"]
+                    )
+
+
+def bgp_vpnv4_table_check_all(router, labelList=None, same=False):
+    """
+    Dump and check that vpnv4 entries are correctly configured with specific label values
+    * 'router': the router to check
+    * 'labelList': check that the label values are not present in the vpnv4 entries
+    *              that list is updated with the present label value found.
+    * 'same': by default, set to False. Addresses groups are classified by addresses.
+    *         if set to True, all entries of all groups should have a unique label value
+    """
+    group1 = ["172.31.0.11/32", "172.31.0.20/32", "172.31.0.111/32"]
+    group2 = ["172.31.0.12/32", "172.31.0.15/32"]
+    group3 = ["172.31.0.13/32"]
+    group4 = ["172.31.0.14/32"]
+    group5 = ["192.168.255.0/24", "192.0.2.0/24"]
+    if same:
+        bgp_vpnv4_table_check(
+            router,
+            group=group1 + group2 + group3 + group4 + group5,
+            labelList=labelList,
+        )
+    else:
+        for group in (group1, group2, group3, group4, group5):
+            bgp_vpnv4_table_check(router, group=group, labelList=labelList)
+
+
+def mpls_table_check(router, blacklist=None, labelList=None, whitelist=None):
+    """
+    Dump and check 'show mpls table json' output. An assert is triggered in case test fails
+    * 'router': the router to check
+    * 'blacklist': the list of nexthops (IP or interface) that should not be on output
+    * 'labelList': the list of labels that should be in inLabel value
+    * 'whitelist': the list of nexthops (IP or interface) that should be on output
+    """
+    nexthopList = []
+    if blacklist:
+        nexthopList.append(blacklist)
+    logger.info("Checking MPLS labels on {}".format(router.name))
+    dump = router.vtysh_cmd("show mpls table json", isjson=True)
+    for entry in dump:
+        assert (
+            "inLabel" in dump[entry].keys()
+        ), "{}, show mpls table, inLabel not found".format(router.name)
```

Comment:
`entry` is always the `inLabel` useless assert

Reply:
done